### PR TITLE
sql: adding missing tables on information_schema

### PR DESCRIPTION
**Pull request by mnovelodou**
Previously, there were 38 undefined information_schema tables that
exists in postgres
This was inadequate because not having them may cause compatibility
issues
To address this, this patch defines these 38 tables as unimplemented
to allow queries

Release note (sql change): adding empty missing tables on information
schema for compatibility:
attributes
check_constraint_routine_usage
column_column_usage
column_domain_usage
column_options
constraint_table_usage
data_type_privileges
domain_constraints
domain_udt_usage
domains
element_types
foreign_data_wrapper_options
foreign_data_wrappers
foreign_server_options
foreign_servers
foreign_table_options
foreign_tables
information_schema_catalog_name
role_column_grants
role_routine_grants
role_udt_grants
role_usage_grants
routine_privileges
sql_features
sql_implementation_info
sql_parts
sql_sizing
transforms
triggered_update_columns
triggers
udt_privileges
usage_privileges
user_defined_types
user_mapping_options
user_mappings
view_column_usage
view_routine_usage
view_table_usage

Related to: #8675

### CODE REVIEW

File: pkg/sql/testdata/mysql_test_expected_diffs_on_information_schema.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
   "pgVersion": "8.0.23",
   "diffSummary": {
     "TotalTables": 79,
-    "TotalColumns": 224,
-    "MissingTables": 57,
-    "MissingColumns": 71,
+    "TotalColumns": 280,
+    "MissingTables": 0,
+    "MissingColumns": 9,
```
Comment (rafiss) [inline pkg/sql/testdata/mysql_test_expected_diffs_on_information_schema.json:7]: what are the 9 missing columns? maybe we could add those too.

**Review by rafiss — CHANGES_REQUESTED** [review]

File: pkg/sql/testdata/mysql_test_expected_diffs_on_information_schema.json (hunk 1 above)
Comment (mnovelodou) [reply pkg/sql/testdata/mysql_test_expected_diffs_on_information_schema.json:7]: I found an issue from this mysql, which seemed outdated, these are the actual values:

```
"diffSummary": {
    "TotalTables": 48,
    "TotalColumns": 280,
    "MissingTables": 21,
    "MissingColumns": 80,
    "DatatypeMismatches": 0
  }
```

**Review by rafiss — APPROVED** [review]
looks good! the PR needs to be rebased

**Pull request merged**